net/modemmanager/files/modemmanager.proto

```diff
@@ -385,6 +385,22 @@ proto_modemmanager_setup() {
 	}
 	echo "modem available at ${modempath}"
 
+	state="$(modemmanager_get_field "${modemstatus}" "state")"
+	state="${state%% *}"
+	if [ "$state" = "locked" ]; then
```

Comment:
How about making a switch case statement here? In my case I have a system that does not have a SIM card and therefore sets the state to `failed`. The value `state-failed-reason` is set to `sim-missing`. Currently it is always trying to configure the interface in an endless loop. 
But this makes no sense, because the SIM is missing.
It would be better to stop now.
With an error:
```
proto_notify_error "${interface}" MM_SIM_MISSING
proto_block_restart "${interface}"
return 1
```